libnavigation-base/src/main/java/com/mapbox/navigation/base/options/OnboardRouterOptions.kt

```diff
@@ -0,0 +1,102 @@
+package com.mapbox.navigation.base.options
+
+import android.content.Context
+
+/**
+ * Defines options for on-board router. These options enable a feature also known as Free Drive.
+ * This allows the navigator to map-match your location onto the road network without a route.
+ *
+ * @param onboardRouterRouterEndpointOptions options for requesting routing tiles
+ * @param tileFilePath File path to store routing tiles
+ *  @see [OnboardRouterEndpointOptions.createDefaultTilePath] for a default storage.
+ * @param inMemoryTileCache Max size of tiles cache (optional)
+ * @param mapMatchingSpatialCache Max size of cache for map matching (optional)
+ * @param threadsCount Max count of native threads (optional)
+ * @param builder used for updating options
+ */
+data class OnboardRouterOptions(
+    val onboardRouterRouterEndpointOptions: OnboardRouterEndpointOptions,
+    val tileFilePath: String,
+    val inMemoryTileCache: Int?,
+    val mapMatchingSpatialCache: Int?,
+    val threadsCount: Int?,
```

Comment:
Not sure if these should be exposed or handled internally (even if we have defaults for them) 🤔

Reply:
threadsCount, mapMatchingSpatialCache, and inMemoryTileCache

agree, and would be happy to remove. that may be out of scope of this change, but would be good to know if we can delete them.

Reply:
Again, I believe this could be an option but definitely something that we should control internally and not expose publicly.

Reply:
Yeah fine by me, I feel the same but was only trying to keep things working the same. Removed! https://github.com/mapbox/mapbox-navigation-android/pull/3020#discussion_r429501058

Reply:
Is it safe @mskurydin? From your point of view, should we expose these?

Reply:
cc @vmihaylenko

Reply:
I think all parameters except inMemoryTileCache (if we ignore the naming) are somewhat arcane to end-users and should not be exposed.

Reply:
ok dug into it a little. in valhalla it's named max_cache_size, looks like the max cache size is 1GB
https://github.com/mapbox/mapbox-navigation-native/blob/c09a6730f78e66ace477c1f1102bbcb1586c3b91/src/mblc/valhalla_conf.cpp#L271

speaking of external naming for end users. In these options they're specifying a filePath for routing tiles. Is this the maxFileSize? Going to set the default value and update the name, let me know what you think

Reply:
Also note that the maximum allowed space to be configured by nav-native RouterParams is `2.147483647` GB. This is because the `inMemoryTileCache` is an Integer

Reply:
oh it's u32, so it's `4.29` GB. Kotlin does have a `UInt` to give a little more space for now.

Reply:
it's a @ExperimentalUnsignedTypes.. 🤔 

would be better as a `Long`

Reply:
advanced users get 4GB by using Int.MIN_VALUE 🙃